src/main/scala/org/broadinstitute/dsde/workbench/sam/Boot.scala

```diff
@@ -86,51 +86,60 @@ object Boot extends App with LazyLogging {
+    def createSamRoutes(cloudExtensions: CloudExtensions, accessPolicyDAO: AccessPolicyDAO): (SamRoutes, UserService, ResourceService, StatusService) = {
+      // TODO - https://broadinstitute.atlassian.net/browse/GAWB-3603
+      // This should JUST get the value from "emailDomain", but for now we're keeping the backwards compatibility code to
+      // fall back to getting the "googleServices.appsDomain"
+      val emailDomain = config.as[Option[String]]("emailDomain").getOrElse(config.getString("googleServices.appsDomain"))
+
+      val resourceService = new ResourceService(resourceTypeMap, accessPolicyDAO, directoryDAO, cloudExtensions, emailDomain)
+      val userService = new UserService(directoryDAO, cloudExtensions)
+      val statusService = new StatusService(directoryDAO, cloudExtensions, 10 seconds)
+      val managedGroupService = new ManagedGroupService(resourceService, resourceTypeMap, accessPolicyDAO, directoryDAO, cloudExtensions, emailDomain)
 
-    val samRoutes = cloudExt match {
-      case googleExt: GoogleExtensions => new SamRoutes(resourceService, userService, statusService, managedGroupService, config.as[SwaggerConfig]("swagger"), directoryDAO) with StandardUserInfoDirectives with GoogleExtensionRoutes {
-        val googleExtensions = googleExt
-        val cloudExtensions = googleExt
+      val samRoutes = cloudExtensions match {
+        case googleExt: GoogleExtensions => new SamRoutes(resourceService, userService, statusService, managedGroupService, config.as[SwaggerConfig]("swagger"), directoryDAO) with StandardUserInfoDirectives with GoogleExtensionRoutes {
+            val googleExtensions = googleExt
+            val cloudExtensions = googleExt
+          }
+        case _ => new SamRoutes(resourceService, userService, statusService, managedGroupService, config.as[SwaggerConfig]("swagger"), directoryDAO) with StandardUserInfoDirectives with NoExtensionRoutes
       }
-      case _ => new SamRoutes(resourceService, userService, statusService, managedGroupService, config.as[SwaggerConfig]("swagger"), directoryDAO) with StandardUserInfoDirectives with NoExtensionRoutes
+      (samRoutes, userService, resourceService, statusService)
     }
 
     for {
-      _ <- schemaDAO.init() recover {
+      _ <- schemaDAO.init().recoverWith{
         case e: WorkbenchException =>
           logger.error("FATAL - could not update schema to latest version. Is the schema lock stuck? See documentation here for more information: [link]")
-          throw e
+          Future.failed(e)
         case t: Throwable =>
           logger.error("FATAL - could not init ldap schema", t)
-          throw t
+          Future.failed(t)
       }
 
-      _ <- Future.traverse(resourceTypes.map(_.name)) { accessPolicyDAO.createResourceType } recover {
-        case t: Throwable =>
-          logger.error("FATAL - unable to init resource types", t)
-          throw t
-      }
+      io = ExecutionContexts.blockingThreadPool.use{
```

Comment:
I think the tests are failing because `blockingThreadPool` gets released when this block completes but _this block completes after successful start_, not when the app is done.